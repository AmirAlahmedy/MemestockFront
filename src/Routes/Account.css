.leftlinks{
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
    color: rgb(0, 121, 211);
    display: flex;
    height: 20px;
    margin-bottom: 4px;
}

.emailParag{
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    color: rgb(124, 124, 124);
}

