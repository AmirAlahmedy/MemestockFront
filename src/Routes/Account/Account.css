.leftlinks{
    font-size: 16px;
    font-weight: 500;
    line-height: 20px;
    color: rgb(124, 124, 124);
    display: flex;
    height: 20px;
    margin-bottom: 4px;
}

.emailParag{
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    color: rgb(124, 124, 124);
}

