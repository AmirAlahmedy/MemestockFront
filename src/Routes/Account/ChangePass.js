import React from 'react';
import {changePasswordApiCall} from '../../actions/Profile'

class ChangePass extends React.Component {
    // constructor(props){
    //     super(props);
    //     this.state = { user: null, oldPassword: '', newPassword: '' };
    //     this.getCurrentUser = this.getCurrentUser.bind(this);
    //     this.handleSubmit = this.handleSubmit.bind(this);
    //     this.handleOldPasswordChange = this.handleOldPasswordChange.bind(this);
    //     this.handleNewPasswordChange = this.handleNewPasswordChange.bind(this);
    // }

    // async getCurrentUser(){
    //     // this.props.auth.getUser()
    //     //     .then(user => this.setState({user}));
    // }

    // handleOldPasswordChange(e) {
    //     this.setState({ oldPassword: e.target.value });
    // }
    // handleNewPasswordChange(e) {
    //     this.setState({ newPassword: e.target.value });
    // }

    // handleSubmit(e){
    //     e.preventDefault();

    //     var data = {
    //         userId: this.state.user.sub, oldPassword: this.state.oldPassword, newPassword: this.state.newPassword
    //     };
    //     this.props.changePasswordApiCall(data);
    // }

    // componentDidMount(){
    //     this.getCurrentUser();
    // }

    render() {
        // if(!this.state.user) return null;
        // const errorMessage = this.props.error ?
        //     (<div className="alert alert-danger"><strong>Error! </strong>{this.props.error}</div>):
        //     null;
        // const successMessage = this.props.success ?
        //     (<div className="alert alert-success"><strong>Success! </strong>{this.props.success}</div>):
        //     null;


        return (
            <div>
                <section className="user-profile">
                    <div>
                        {/* <label>Name:</label>
                        <span>{this.state.user.name}</span> */}
                    </div>
                </section>
                <br />
                <br />
                <form className="form-horizontal col-sm-6" onSubmit={this.handleSubmit}>
                <h3>Change Password</h3>
                    <br />
                    {/* {errorMessage}
                    {successMessage} */}
                    <div className="form-group">
                        <label>Old Password:</label>
                        <input className="form-control" type="password" id="oldPassword" autoComplete="current-password"
                               onChange={this.handleOldPasswordChange} />
                    </div>
                    <div className="form-group">
                        <label>New Password:</label>
                        <input className="form-control" type="password" id="newPassword"  autoComplete="new-password"
                               onChange={this.handleNewPasswordChange} />
                    </div>
                    <input className="btn btn-outline-success col-2" type="submit" id="submit" value="Submit"/>
                </form>
            </div>
        )
    }
};


const mapStateToProps = (state) => {
    return {
        error: state.profile.error,
        success: state.profile.success
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        changePasswordApiCall: (data) => dispatch(changePasswordApiCall(data))
    }
};

export default ChangePass;