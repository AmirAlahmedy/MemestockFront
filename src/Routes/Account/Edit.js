import React, { Component } from 'react'
import axios from '../../axios-orders';
class Edit extends Component {

  state = {
    email: ''
  }
 /**
 * Gets the current  user name from the local storage
 * @function getCurrentUser
 */
  getCurrentUser() {
    return localStorage.getItem("Username");
  }

  /**
 * Handles Email change.
 * @function handleChange
 * @param {event} e - The change of the Email.
 */
  handleChange = (e) => {
    this.setState({
      email: e.target.value
    })
  }
  /**
 * Handles edit-Email form submission.
 * @function handleSubmit
 * @param {event} e - The submission of the edit-Email form.
 */
  handleSubmit = (e) => {
    e.preventDefault();
    let username = this.getCurrentUser();
    alert(username);


    let checker = "";

    let Emailcontent = document.getElementById("userEmail").value;

    if (Emailcontent === checker) {
      alert("Please provide an Email !");
      return;
    }
    var headers = {
      'auth': localStorage.getItem("token")
    }
    var data = { Email: this.state.email }

    console.log(data);
    axios.put('/me/edit/email/' + username, data, { headers: headers })
      .then(res => {
        console.log(res);
        if (res.status == 200) {
          alert("Email changed Successfully!");
        } else if (res.status === 401 || res.status === 404) {
          alert("Email  Unsuccessful");
          return Response.json;
        }
      })
      .catch(error => {
        alert("Error Caught");
      })
  }


  render() {
    return (
      <div className="edit">

        <form onSubmit={this.handleSubmit}>

          <div className="form-group">
            <label>E-Mail
              <input class="form-control" type="Email" placeholder="Update E-mail" id="userEmail" onChange={this.handleChange} />
            </label>
          </div>
          <input class="btn" type="submit" value="Update" />
        </form>

      </div>
    )
  }
}

export default Edit
