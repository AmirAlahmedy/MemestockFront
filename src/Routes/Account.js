import React from 'react'
import './Account.css'

const Account = () => {
  return (
    <div>
      <div className="leftlinks">
     {/* <h4 className="left">Account Settings</h4> */}

    <div><a href="#" aria>Change/add Email</a></div>
  
    <div><a href="#">Change Password</a></div>
    
        </div>

        
    </div>
  )
}

export default Account
