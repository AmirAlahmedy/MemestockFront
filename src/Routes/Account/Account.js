import React, { Component } from 'react';
import {Link,Route} from 'react-router-dom';
import './Account.css'
import Edit from './Edit'

class Account extends Component {
  render(){
  return (
    <div>
      <div >
     {/* <h4 className="left">Account Settings</h4> */}

    <div className="leftlinks"><Link to ='/Settings/Account/Edit/'>Change/add Email</Link></div>
    <p className="emailParag">The Above Link is not working yet GUYS Stay Tunedd!</p>
   
    <Route path='/Settings/Account/Edit'  component={Edit}/>
   
    <div className="leftlinks"><Link to="/Settings/Account/">Change Password</Link></div>
    <p className="emailParag">Change password is still oUnder Construction. We're in the process of building this functionality. Stay tuned!</p>
        </div>

       
    </div>
  )
}
}

export default Account
