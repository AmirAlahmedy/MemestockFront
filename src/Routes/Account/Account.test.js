import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import Account from './Account';
import { create } from "react-test-renderer";

configure({adapter: new Adapter()});
//jest.mock("axios");

describe('<Account/>', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Account></Account>);
    });

    test("it matches the snapshot", () => {
        const component = create(<Account />);
        expect(component.toJSON()).toMatchSnapshot();
      });

    it("Password ", async () => {
        const component = create(<Account />);
        const instance = component.getInstance();
     //   await instance.componentDidMount();
        console.log(instance.state) 
    });
})