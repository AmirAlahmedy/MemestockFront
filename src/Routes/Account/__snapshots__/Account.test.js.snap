// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account/> it matches the snapshot 1`] = `
<div>
  <div>
    <div
      className="accountLinks"
    >
      <a
        href="#Edit"
      >
        Update Email
      </a>
      <a
        href="#ChangePass"
      >
        Change Password
      </a>
    </div>
  </div>
  <div
    className="edit"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          E-Mail
          <input
            class="form-control"
            id="userEmail"
            onChange={[Function]}
            placeholder="Update E-mail"
            type="Email"
          />
        </label>
      </div>
      <input
        class="btn"
        type="submit"
        value="Update"
      />
    </form>
  </div>
</div>
`;
